compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -2285,9 +2285,31 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
 
                 let source_map = self.r.session.source_map();
 
+                // Make sure this is actually crate-relative.
+                let use_and_crate = import.use_span.with_hi(after_crate_name.lo());
+                let is_definitely_crate =
+                    source_map.span_to_snippet(use_and_crate).map_or(false, |s| {
```

Comment:
Could you use `ImportKind` to check if the import is crate relative or not? I think this can also remove the use of `span_to_snippet`sanipulating strings.